tests/CMakeLists.txt

```diff
@@ -184,8 +185,15 @@ IF (NOT DISABLE_TESTS)
 				   tests.cpp
 				   ${OPENSHOT_TEST_FILES} )
 
-	# Link libraries to the new executable
-	target_link_libraries(openshot-test openshot ${UNITTEST++_LIBRARY})
+	IF (NOT DISABLE_EXTERNAL_EFFECTS)
+		# Link libraries to the new executable
+		target_link_libraries(openshot-test openshot ${UNITTEST++_LIBRARY} effect-superblur)
+		add_definitions(-DTEST_PLUGIN="${PROJECT_BINARY_DIR}/external-effects/super-blur/libeffect-superblur.so"
+						-DTEST_PLUGIN_NAME="SuperBlur")
+	ELSE()
+		# Link libraries to the new executable
+		target_link_libraries(openshot-test openshot ${UNITTEST++_LIBRARY})
+	ENDIF(NOT DISABLE_EXTERNAL_EFFECTS)
```

Comment:
Couple issues here:

1. `DynamicEffects_Tests.cpp` will be in the test target sources list even when `DISABLE_EXTERNAL_EFFECTS` is set. That's _okaaaay_ technically, because the whole file's wrapped in `#ifdef TEST_PLUGIN`, but if nothing else it's an extra source file being compiled for no reason.

2. Bigger issue, if you want to test the `dlopen()` functionality of `EffectInfo::LoadEffect`, I suspect you can't _link_ the executable with the library you're trying to `dlopen()` or it will invalidate the test.

   Because `effect-superblur` is added to the `target_link_libraries()` of `openshot-test`, it'll be dynamically linked with the executable:
   ```console
   $ cd build
   $ make openshot-test
   $ ldd tests/openshot-test |grep -i effect
	libeffect-superblur.so => /home/ferd/rpmbuild/REPOS/libopenshot/build/external-effects/super-blur/libeffect-superblur.so (0x00007f41707c6000)
   ```
   The dynamic loader is sure to be smart enough that, when you call `dlopen()` on that file path, it's not going to load anything at all — it's just going to return a handle to the library that was **already** loaded at the start of program execution.

   To test dynamic-loading from the library file on disk, you _don't_ want to link the executable to it. It should have no knowledge of that library prior to `dlopen()` being given the pathname to load it from.

I guess my preference would be something more like this:
```suggestion
# Link libraries to the new executable
target_link_libraries(openshot-test openshot ${UNITTEST++_LIBRARY})
if (NOT DISABLE_EXTERNAL_EFFECTS)
    target_sources(openshot-test PRIVATE DynamicEffects_Tests.cpp)
    target_compile_definitions(openshot-test PRIVATE
        TEST_PLUGIN="${PROJECT_BINARY_DIR}/external-effects/super-blur/libeffect-superblur.so"
        TEST_PLUGIN_NAME="SuperBlur")
endif ()
```

That'll add in the source file and the definitions iff enabled, and leave the library unlinked so it'll _have_ to be loaded from the path in the `TEST_PLUGIN` definition.

Reply:
I made compilation of file `DynamicEffects_Tests.cpp` optional, also I remove static linking with effect library.
@ferdnyc could you, please, have a look?

Reply:
Wow, you are _OLD_-school CMake, huh? :laughing: Well, that's cool, whatever works — and this does, thanks. One day I'll drag this project kicking and screaming into the [Modern CMake](https://cliutils.gitlab.io/modern-cmake/) world, but that day is not today.

I'm actually a little surprised `add_definitions()` is still effective even when it comes after the creation of the executable target (as opposed to `target_compile_definitions()` operating on the target), but it appears that's the case.

> I made compilation of file `DynamicEffects_Tests.cpp` optional, 

And the `add_dependency()` is a good idea, I missed that.

> also I remove static linking with effect library.

*dynamic linking. (A bit pedantic, I admit, but the distinction is critical. Static linking compiles the library right into the executable and creates no external dependencies. With a shared library the runtime dependency is still dynamic linking — as opposed to dynamic _loading_ via `dlopen()`, which is what you're doing instead, and requires no prior linkage between the executable and the shared object file.)